File: ZkTony-Manager/app/src/main/kotlin/com/zktony/manager/data/remote/model/Product.kt
package com.zktony.manager.data.remote.model

import com.zktony.www.common.extension.currentTime
import java.util.UUID

/**
 * @author: 刘贺贺
 * @date: 2023-02-16 16:19
 */
data class Product(
    val id: String = UUID.randomUUID().toString(),
    val software_id: String = "",
    val customer_id: String = "",
    val equipment_id: String = "",
    val express_number: String = "",
    val express_company: String = "",
    val equipment_number: String = "",
    val equipment_time: String = "",
    val attachment: String = "",
    val remarks: String = "",
    val create_by: String = "",
    val create_time: String = currentTime()
)

File: ZkTony-Api/database/src/models/manager/product.rs
use serde::Deserialize;

#[derive(Deserialize, Debug, Clone)]
pub struct ProductSaveReq {
    pub id: String,
    pub software_id: String,
    pub customer_id: String,
    pub equipment_id: String,
    pub express_number: String,
    pub express_company: String,
    pub equipment_number: String,
    pub equipment_time: String,
    pub attachment: String,
    pub remarks: Option<String>,
    pub create_by: Option<String>,
    pub create_time: Option<String>,
}

#[derive(Deserialize, Debug, Clone)]
pub struct ProductDeleteReq {
    pub id: String,
}

#[derive(Deserialize, Debug, Clone)]
pub struct ProductGetReq {
    pub id: Option<String>,
    pub software_id: Option<String>,
    pub customer_id: Option<String>,
    pub equipment_id: Option<String>,
    pub express_number: Option<String>,
    pub express_company: Option<String>,
    pub equipment_number: Option<String>,
    pub create_by: Option<String>,
    pub begin_time: Option<String>,
    pub end_time: Option<String>,
}
